common/interfaces/card_interface/nfc_events.c

```diff
@@ -0,0 +1,221 @@
+ * grant to you, the right to Sell the Software.
+ *
+ * For purposes of the foregoing, "Sell" means practicing any or all
+ * of the rights granted to you under the License to provide to third
+ * parties, for a fee or other consideration (including without
+ * limitation fees for hosting or consulting/ support services related
+ * to the Software), a product or service whose value derives, entirely
+ * or substantially, from the functionality of the Software. Any license
+ * notice or attribution required by the License must also include
+ * this Commons Clause License Condition notice.
+ *
+ * Software: All X1Wallet associated files.
+ * License: MIT
+ * Licensor: HODL TECH PTE LTD
+ *
+ ******************************************************************************
+ */
+
+/*****************************************************************************
+ * INCLUDES
+ *****************************************************************************/
+#include "nfc_events.h"
+
+#include "app_error.h"
+#include "memzero.h"
+#include "nfc_events_priv.h"
+#include "string.h"
+/*****************************************************************************
+ * EXTERN VARIABLES
+ *****************************************************************************/
+
+/*****************************************************************************
+ * PRIVATE MACROS AND DEFINES
+ *****************************************************************************/
+#define DEFAULT_CARD_REMOVAL_RETRY_COUNT 5
+
+/*****************************************************************************
+ * PRIVATE TYPEDEFS
+ *****************************************************************************/
+
+/*****************************************************************************
+ * STATIC VARIABLES
+ *****************************************************************************/
+static nfc_task_states_t nfc_state;
+static nfc_event_t nfc_event;
+static uint8_t card_removal_retry_counter = 0;
+
+/*****************************************************************************
+ * GLOBAL VARIABLES
+ *****************************************************************************/
+
+/*****************************************************************************
+ * STATIC FUNCTION PROTOTYPES
+ *****************************************************************************/
+
+/**
+ * @brief   Checks if card has been removed and returns status
+ * @arg     retry_count:  No. of consecutive retries for card not detected
+ *
+ * @return  true if card is not detected for consecutive retries exceeding
+ * retry_count
+ */
+static bool check_card_removed_status(uint8_t retry_count);
+
+/**
+ * @brief   Handles response from read NFC A target init api
+ *          If card is selected, card select event is set, if an error is
+ * received, other than NFC_RESP_NOT_READY, state is updated to
+ * NFC_STATE_SET_SELECT_CARD_CMD
+ */
+static void nfc_handle_card_select_resp();
```

Comment:
Let's add `void` as function parameter if the function does not expect any input. Applicable to all new functions.
Therefore, this function becomes: `static void nfc_handle_card_select_resp(void);`

For reference: https://app.clickup.com/9002019994/v/dc/8c8zv4u-20804/8c8zv4u-27384

Reply:
updated